### DISCUSSION TITLE
Deye Battery Offline in Cloud - Increase the poll interval for WiFi logger to 10+ seconds

### DISCUSSION THREAD

**Discussion by OwlBawl** [davidrapan/ha-solarman#208] [Q&A] [answered]
I have problems with concurent connection to WiFi logger using Deye Cloud and Ha-solarman
With default Cloud update interval and Ha-solarman - my battery become "offline" shortly after integration start.
When disable it - Deye cloud battery ok again.
StephanJoubert's integration don't overload logger as this one. But it has ~30s update interval as i know.
The rs485 integration by kellerza with 5s update is also ok.
So, the only problem I see - serial port overflow with concurent requests...

**Comment by davidrapan** [comment]
You can change it by using custom profile and define the intervals there.

Edit: Btw you are using all three of them at the same time? 🤯

**Reply by OwlBawl** [reply]
Would appreciate for little help.
Does any one else faced such a problem? And what would be "safe" poll interval to preserve Cloud working?

Logger SN starts with #29

**Reply by OwlBawl** [reply]
> Edit: Btw you are using all three of them at the same time? 🤯

I've started with Cloud and this Sunsynk integration. But got faults I've described. So found an alternative and connected modbus wired (sunsynk was shut down so).

But I'm interested in finding solution to be able use 2 of them in final. 

Want to leave native Cloud active for setup, and smth local for my home integration.

**Reply by davidrapan** [reply]
The thing is that even 1 second should be "safe" and not just cause it should handle it but because the stick will just refuse to answer when you hit it during the time when it's uploading data to the solarman servers (it has precedence).

Just copy profile you are using to to the `inverter_definitions/custom/*.yaml`, open it and change all occurrences of `update_interval: 5` to let's say `update_interval: 10`.

And then configure your device again using this profile.

**Reply by OwlBawl** [reply]
That's strange so, because it somewhy made my Deye battery "offline" in Cloud while using default hybrid.yaml

But have to notice - in integration battery was shown as "Pylontech" if I remember correctly, but definitely not Deye with 00 BMS setting.

The inverter and logger itself remained "online"

**Reply by davidrapan** [reply]
> But have to notice - in integration battery was shown as "Pylontech" if I remember correctly, but definitely not Deye with 00 BMS setting.

Pylontech in this case is rather a name for the protocol used.

> That's strange so, because it somewhy made my Deye battery "offline" in Cloud while using default hybrid.yaml
The inverter and logger itself remained "online"

Whoa, I'm getting confused, tell me more about the setup you have...

**Reply by OwlBawl** [reply]
With pleasure, what info do you need?

Inverter - Deye SUN-6K-SG03LP1-EU
Logger WiFi with SN #29*
Battery - Deye RW-M5.3 Pro

**Reply by OwlBawl** [reply]
Batt controller stayed online (green mark)
While battery itself became offline (red mark)

![image](https://github.com/user-attachments/assets/9f74ca45-e0c0-47b5-aa3e-1df35c3da4e7)

**Reply by davidrapan** [reply]
No I mean what does it mean battery offline? Do you have in the HA inverter and battery separately using solarman protocol or?

**Reply by davidrapan** [reply]
Okay so in the cloud you can see the inverter and battery separately but they are connected using single stick?

**Reply by OwlBawl** [reply]
Ofcourse, Deye logger reads data thrue inverter. It's usual behaviour

**Reply by davidrapan** [reply]
What? I don't think we understand each other. Seeing inverter and also the battery is not usual behavior. That's what I'm talking about and what got me confused. 😉

**Reply by OwlBawl** [reply]
Strange, not first Deye inverter I see with battery data in cloud. Maybe I'm not so experienced with them of course

**Reply by davidrapan** [reply]
Maybe with the combination of Deye Inverter with Deye Battery.

**Reply by OwlBawl** [reply]
Yes, exactly, I saw it with Deye-Deye set on other variations

**Reply by OwlBawl** [reply]
So.... in search of unconflicted solution with poll interval )
Anyway interesting to figure it out. For now I know that you default interval (5s?) fils battery communication, while Stephan's integration with 30s interval works fine, but think it's to long

**Reply by davidrapan** [reply, marked as answer]
I told you, try to increase it to 10. [Here is how](https://github.com/davidrapan/ha-solarman/discussions/208#discussioncomment-10922290)

**Reply by OwlBawl** [reply]
Would try, thank you.
Was a bit exited that this "problem" is undiscovered by anyone before.

**Reply by davidrapan** [reply]
If there is an issue I'm pretty sure it will be about congestion. When all of the events happens at the same time which means that lower polling rate will lower the chance of it happening but won't remove the issue completely.

**Reply by OwlBawl** [reply]
So, maybe my current rs485 integration would be more stable and fault safe

**Reply by davidrapan** [reply]
I mean you should for sure prefer Sunsynk addon over this integration if you have the chance. 😉

**Reply by oxynic** [reply]
> Would try, thank you. Was a bit exited that this "problem" is undiscovered by anyone before.

I had same problem with Deye - Deye, i increased the polling to 15sec

**Comment by PV-Joe** [comment]
I recommend to use ha-solarman only as it provides all necessary information.

Or is there any other values by the other two integrations that are not taken into account by ha-solarman?

**Reply by OwlBawl** [reply]
Deye's WiFi definitely not the most reliable thing I met )
Thing wired connection it always priority for critical hardware

**Reply by davidrapan** [reply]
I would not call it Deye WIFI.

**Reply by PV-Joe** [reply]
ha-solarman by David Rapan runs since month 100% stable without any problems, so I can't confirm that the LSW3 Stick-Logger is having any problems with stability.

It is pulling the "current sensors" every 5 seconds from my Deye Hybrid.

Maybe it is a WiFi problem that could be optimized?

**Reply by OwlBawl** [reply]
I have separate 2,4Ghz network for all my iot things working just fine on UniFi.
This one work also, but definitely know that things like "bind to ap" make problems to Deye logger, it could have issues on reconnect, so canceled it. Anyways, WiFi vs wire for me personally is obvious for stability.

**Reply by davidrapan** [reply]
> WiFi vs wire for me personally is obvious for stability.

Not sure about the state with WiFi 7 but in overall I agree for sure!

**Comment by OwlBawl** [comment]
Made custom yaml
For Deye hybrid - update interval 10 instead of 5 seems to be working. Battery disconnection not mentioned yet.

**Reply by FrancescoCosta78** [reply]
Hi, I tried to put as you indicate 10 

info:
manufacturer: Deye
model: SG0XLP1

default:
update_interval: **10** 
digits: 6

- group: PV
update_interval: **10** 
items:


but the problem remains...

Did you change anything else?

**Reply by OwlBawl** [reply]
No, as I remember. Try 15 sec.
And check if any sensor update interval set frequently.
As for me I've moved to cabled modbus

**Reply by Lanzarhack** [reply]
Where is this writen this option. Cant find it to change the value. New to HS so bare with me. :)

**Reply by davidrapan** [reply]
You can for example install File editor addon and then in `/config/custom_components/solarman/inverter_definitions/deye_p3.yaml`

**Reply by Lanzarhack** [reply]
Thank you. One more question. IF i disable all the enities under deye that i dont need. Will this also skip and reduce the identites that HA reads from the loger. Since if i have 202 identities and only need to read 80 this shuld help and reduce the read interval and trafic.
for example picture if i disable the ones that i dont need.
![Screenshot 2025-05-13 10 10 33](https://github.com/user-attachments/assets/a3433235-6013-409f-aff2-46bc835fc908)

**Reply by davidrapan** [reply]
No it will not because it doesn't request all the sensors all the time. It has scheduler and most sensors are requested once 5m. Only real-time sensors such as Power, Voltage, Current or Battery are requested often but they all fit into single request to the inverter (so it doesn't really matter if it's one or hundred here). And Battery X (X is number) can be disabled in device configuration > Additional options > Number of Battery packs set to 0 (default is -1 which is auto).

**Reply by FrancescoCosta78** [reply]
I have now put "update interval: 25" in all references. I only have battery number 1 offline every now and then, while inverter, logger, battery number 0 is online.

I have defined a custom file, but each update deletes it and I have to put it back from the backup

**Reply by davidrapan** [reply]
Did you try disable additional battery reading in device configuration > Additional options > Number of Battery packs set to 0?

**Reply by FrancescoCosta78** [reply]
My Config:
![image](https://github.com/user-attachments/assets/8b843ea8-fc23-4e4b-a94a-6dcd79886401)

**Reply by davidrapan** [reply]
So? Try to set Battery to 0. And why do you have phase set to 1?

**Reply by FrancescoCosta78** [reply]
Single phase system
here is the configuration.

This is how normally everything is ok the cloud tells me
![image](https://github.com/user-attachments/assets/29d7d991-f5a1-4eed-a41b-e9214d7173ac)
![image](https://github.com/user-attachments/assets/74bda1d9-0cf5-445a-9926-61876d3f9bcf)
![image](https://github.com/user-attachments/assets/8b20f19a-7fea-404b-ab81-f3535a4e5f48)

How do I set it to be ok?

**Reply by Lanzarhack** [reply]
SO if my model off inverter is for deye_p3.yaml i should change auto for deye_p3.yaml and then under batery pack i should deselect the number off batery packs and it should ignore it.

![Screenshot 2025-05-13 10 48 02](https://github.com/user-attachments/assets/621076bf-77f0-4316-a613-ee7132b5c20d)

**Reply by davidrapan** [reply]
You don't need to change the profile (it won't do anything anyway, all deye profiles activates autodetection 😉) Yes, you have currently batteries set to -1 which is auto... move it one to the right to 0.

**Reply by Lanzarhack** [reply]
OK set it to 0 and the batery in deye app still shows offline most off the time. Any other suggestions.

**Reply by FrancescoCosta78** [reply]
SET to ZERO all offline.

![image](https://github.com/user-attachments/assets/80abab4f-afdf-4c80-8264-54ba1d3795ff)

**Reply by davidrapan** [reply]
This shouldn't have be solution (it should just help a little). You will proly still need to increase that base update interval to 10 s or something.

**Reply by FrancescoCosta78** [reply]
Modify:


default:
  update_interval: 25         #(OLD 5)
  digits: 6


  - group: Control
    update_interval: 25         #(OLD 5)
    items:

  - group: PV
    update_interval: 25         #(OLD 5)
    items:

It's not ok definityve....

Idea?

**Reply by davidrapan** [reply]
I mean this and disable reading of additional battery sensor should be enough. It's what's always helped so far.

**Reply by FrancescoCosta78** [reply]
if insert into group battery ?
![image](https://github.com/user-attachments/assets/4aa85e53-d0aa-4b88-9f65-f2c318e6efb5)

**Comment by FrancescoCosta78** [comment]
Isn't it by chance the request to some register that gives some error?
In the log there is nothing with the writing err or error or ko.
The other integration was set to default 5 and never blocked. True that it didn't have all these sensors...

Anyway I put all the intervals that were at 5 -> 15 and let's see.

**Comment by trozman** [comment]
Same problem here (Inv: Deye, batt: Deye). Then I edited in /solarman/inverter_definitions/deye_hybrid.yaml (using Studio Code Server HA addon):
...
default:
  update_interval: 10 #old: 5

...

And deye cloud app/devices/battery stays online (observing for 30 minutes). Will observe if stays that way**

I also tried adding update_interval to Battery group:
- group: Battery
    update_interval:10 #added

but it didn't have an effect.

****update 1: no, increasing to 10s didn't work. Increasing poll interval and observing when it's stable...**
****update 2: 15s interval mostly works ok, but I had to change const.py: TIMINGS_INTERVAL_SCALE = 3.**
****update 3 (4. 3. 2026): after the firmware update (to include 'Smart devices'), I had to increase the interval TIMINGS_INTERVAL_SCALE = 3  (20 secs).**

**Comment by techadrian** [comment]
Hi,
I had the same issue with partial offline in Deye Cloud/Solarman for the Battery status being offline.

Considering what was discussed above, I have done a replacement of all "update_interval: 5" values to "update_interval: 21" and solved my issue. I am using deye_p3.yaml profile. 

I do not rely on such tight pull intervals, so most probably doing 30 sec intervals would offer the best stability for all platforms.

**Comment by schumixmd** [comment]
Can we add Default Update Interval as a parameter? Instead of messing up with the profiles?
The reason is that I like the 5sec pooling interval, but if I want to change some parameters from the cloud, most of the time it does not work. So, changing the parameter in HA, would be way easier than anything else.
